keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/nuts/Stepsize.java

```diff
@@ -146,10 +149,11 @@ private double updateLogStepSize(Tree tree, int sampleNum) {
         //(1-m^-k) * log(epsilon_bar_m-1)
         double increasedStepSizeFrozen = (1 - tendToZero) * logStepSizeFrozen;
 
+        //log(epsilon_bar_m) = m^-k * log(epsilon_m) + (1 - m^-k) * log(epsilon_bar_m-1)
+        logStepSizeFrozen = reducedStepSize + increasedStepSizeFrozen;
         averageAcceptanceProb = updatedAverageAcceptanceProb;
 
-        //log(epsilon_bar_m) = m^-k * log(epsilon_m) + (1 - m^-k) * log(epsilon_bar_m-1)
-        return reducedStepSize + increasedStepSizeFrozen;
+        return updatedLogStepSize;
```

Comment:
These all look like reasonable changes - but I'll read through the paper again and compare to be absolutely sure.

Reply:
Yes, please. The paper is a beast if you've never looked at it. Let me know if you'd like a summary before you dig in.

Reply:
I've read it once before and thought I'd understood it, but I'll grab you if I'm struggling!  I just want to walk through side by side with the code to make sure I agree it's all looking good.